src/builder/rollup.ts

```diff
@@ -179,24 +179,47 @@ export async function rollupBuild(ctx: BuildContext) {
     await ctx.hooks.callHook("rollup:dts:options", ctx, rollupOptions);
     const typesBuild = await rollup(rollupOptions);
     await ctx.hooks.callHook("rollup:dts:build", ctx, typesBuild);
+    // #region cjs
+    if (ctx.options.rollup.emitCJS) {
+      await typesBuild.write({
+        dir: resolve(ctx.options.rootDir, ctx.options.outDir),
+        entryFileNames: "[name].d.cts",
+        chunkFileNames: (chunk) => getChunkFilename(ctx, chunk, "d.cts"),
+      });
+    }
+    // #endregion
+    // #region mjs
+    await typesBuild.write({
+      dir: resolve(ctx.options.rootDir, ctx.options.outDir),
+      entryFileNames: "[name].d.mts",
+      chunkFileNames: (chunk) => getChunkFilename(ctx, chunk, "d.mts"),
+    });
+    // #endregion
+    // #region .d.ts for node10 compatibility
```

Comment:
@pi0 @antfu @danielroe Should I add a option to enable/disable generate `.d.ts`

Reply:
I think probably the backwards compatibility mode should be default. It won't disadvantage newer module resolution, right?

Reply:
According to @antfu's suggestion, I added `"compatible" | "node16"` to the declaration field